.github/workflows/infracost.yaml

```diff
@@ -0,0 +1,60 @@
+name: Infracost exibe estimativa de custo da infraestrutura
+
+on:
+  workflow_call:
+    inputs:
+      OS_VERSION:
+        description: "Versão do sistema operacional"
+        required: false
+        default: "ubuntu-latest"
+        type: string
+      TF_DIRECTORY:
+        description: "Diretório do projeto terraform"
+        required: false
+        default: .
+        type: string
+    secrets:
+      INFRACOST_API_KEY:
+        description: "APIKEY do Infracost"
+        required: true
+jobs:
+  infracost:
+    name: Infracost workflow
+    runs-on: ${{ inputs.OS_VERSION }}
+    env:
+       INFRACOST_ENABLE_CLOUD: true
+    steps:
+      - name: Configura Infracost
+        uses: infracost/actions/setup@v2
+        with:
+          api-key: ${{ secrets.INFRACOST_API_KEY }}
+
+      - name: Checkout base branch
+        uses: actions/checkout@v2
+        with:
+          ref: '${{ github.event.pull_request.base.ref }}'
+
+      - name: Gerando arquivo Infracost custo estimado
+        run: |
+          infracost breakdown --path=${{ inputs.TF_DIRECTORY }} \
+                              --format=json \
+                              --out-file=/tmp/infracost-base.json
+
+      - name: Checkout PR branch
+        uses: actions/checkout@v2
+
+      # Generate an Infracost diff and save it to a JSON file.
+      - name: Gerando diff Infracost
+        run: |
+          infracost diff --path=${{ inputs.TF_DIRECTORY }} \
+                          --format=json \
+                          --compare-to=/tmp/infracost-base.json \
+                          --out-file=/tmp/infracost.json
+
+      - name: Adicionando comentário no PR
+        run: |
+            infracost comment github --path=/tmp/infracost.json \
+                                     --repo=$GITHUB_REPOSITORY \
+                                     --github-token=${{github.token}} \
```

Comment:
Seria bom testar, mas acho que esse token não vai ter permissão de comentar se o workflow for chamado de um PR que é um fork de uma pessoa que não é maintainer do repo.

Reply:
@lgfa29 eu fiz a validação aqui [rds](https://github.com/EzzioMoreira/rds-terraform/pull/3) e funcionou como esperado. O [github-token](https://docs.github.com/pt/actions/security-guides/automatic-token-authentication) tem permissão de leitura/escrita no repositório corrente.

Reply:
A permissão do token varia dependendo da pessoa que ativou a ação. Como você tem permissão de escrita no repo, os seus PRs vão funcionar, na tabela de permissão tem que ver a última coluna, que é para repositórios criados a partir de um fork:

![image](https://user-images.githubusercontent.com/775380/185794888-45293f0f-2223-437b-acbb-c14e8e3fa21d.png)


Eu criei um [PR de teste](https://github.com/EzzioMoreira/rds-terraform/pull/6) no seu repo, vamos ver se funciona :smile:

Reply:
Falamos na ultima reunião de colocar isso em outro lugar, mas seria a partir da main, né? Dai não precisaria desse token, né?